spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/health/SpanCollectionHealthIndicator.java

```diff
@@ -0,0 +1,26 @@
+package org.springframework.cloud.sleuth.health;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.actuate.health.AbstractHealthIndicator;
+import org.springframework.boot.actuate.health.Health;
+
+/**
+ * Health indicator that shows the number of accepted / dropped spans.
+ *
+ * //TODO: should this be a health indicator (it's never down) or an Endpoint ?
```

Comment:
mentioned elsewhere, but I totally agree that healthcheck is a different concern, or at least one harder to pin down with a ratio of accept vs drop without any history.

Local, health checks are usually something you can answer yes/no to. For example, is the client busted? One way is to inject the transport itself (ex the reporter) and expose a state of the connection.
https://github.com/Netflix/suro/blob/master/suro-server/src/main/java/com/netflix/suro/server/HealthCheck.java

@kristofa any thoughts on this?